src/Types/Unify.hs

```diff
@@ -36,6 +36,12 @@ bind var ty
 unify :: Type Typed -> Type Typed -> SolveM ()
 unify (TyVar a) b = bind a b
 unify a (TyVar b) = bind b a
+
+unify t@TySkol{} b
```

Comment:
Why this rather than
```haskell
unify TySkol{} TySkol{} = pure ()
unify t@TySkol{} b = throwError $ SkolBinding t b
unify b t@TySkol{} = throwError $ SkolBinding t b
```

I know it's more cases, I guess I just find it cleaner.